openedx/core/djangoapps/team_api/views.py

```diff
@@ -0,0 +1,407 @@
+            to HTML escape them to avoid script injections, as the data is
+            stored exactly as specified. The intention is that plain text is
+            supported, not HTML.
+
+            If the user is not logged in, a 403 error is returned.
+
+            If the specified team does not exist, a 404 error is returned.
+
+        **Response Values for PATCH**
+
+            Only staff can patch teams. If the specified team does not exist, a
+            404 error is returned.
+
+            If "application/merge-patch+json" is not the specified content type,
+            a 415 error is returned.
+
+            If the update could not be completed due to validation errors, this
+            method returns a 400 error with all error messages in the
+            "field_errors" field of the returned JSON.
+    """
+
+    authentication_classes = (SessionAuthenticationAllowInactiveUser, OAuth2AuthenticationAllowInactiveUser)
+    permission_classes = (permissions.IsAuthenticated, IsStaffOrReadOnly)
+    lookup_field = 'team_id'
+    serializer_class = CourseTeamSerializer
+    parser_classes = (MergePatchParser,)
+
+    def get_queryset(self):
+        """
+        Returns the queryset used to access the given team.
+        """
+        return CourseTeam.objects.all()
+
+
+class TopicListView(APIView):
+    """HTTP endpoint for retreiving a list of topics for a given course."""
+
+    authentication_classes = (SessionAuthenticationAllowInactiveUser,)
+    permission_classes = (permissions.IsAuthenticated,)
+
+    page_size = 10
+    # TODO determine the correct max page size
+    max_page_size = sys.maxint
+
+    def get(self, request):
+        """
+        GET /api/team/v0/topics/?course_id={course_id}
+        """
+        try:
+            course_id_string = request.QUERY_PARAMS.get('course_id', None)
+            if course_id_string is None:
+                return Response(status=status.HTTP_400_BAD_REQUEST)
+
+            course_id = CourseKey.from_string(course_id_string)
+
+            course_module = modulestore().get_course(course_id)
+            if course_module is None:  # course is None if not found
+                return Response(status=status.HTTP_404_NOT_FOUND)
+
+            if CourseEnrollment.get_enrollment(request.user, course_id) is None:
+                return Response({'detail': "user must be enrolled"}, status=status.HTTP_403_FORBIDDEN)
+
+            topics = [t for t in course_module.teams_topics if t['is_active']]
+
+            if 'text_search' in request.QUERY_PARAMS:
+                return Response({'detail': "text_search is not yet supported"}, status=status.HTTP_400_BAD_REQUEST)
+
+            ordering = request.QUERY_PARAMS.get('order_by', 'name')
+            if ordering == 'name':
+                topics = sorted(topics, key=lambda t: t['name'])
+            elif ordering == 'team_count':
+                topics = sorted(topics, cmp=lambda t1, t2: t1['team_count'] > t2['team_count'])
```

Comment:
team_count will not be stored in the course teams metadata. It will have to be computed. For now, leave it out since implementing that is really part of the sorting/filtering epic (https://openedx.atlassian.net/browse/TNL-1932).